src/evm/protocol/uniswap_v4/state.rs

```diff
@@ -217,50 +227,92 @@ impl UniswapV4State {
     }
 }
 
-#[allow(unused_variables)] //TODO: remove when implemented
 impl ProtocolSim for UniswapV4State {
+    // Not possible to implement correctly with the current interface because we need to know the
+    // swap direction.
```

Comment:
Are we going to not implement it at all? Where is it used? If needed we can return some sort of best guess - like a minimum fee?

Reply:
It's not used AFAIK, this one isn't implemented for VMs also. We need to find a way to make this interface better, but in the scope of this epic it's not planned to be implemented no. 

The minimum fee was also suggested by Alan, but I find it unexpected from a client POV so I decided to leave it todo for now. 
We can easily come back to this later when they launch, for example if they don't enable any protocol fee (similar to uniswapv3) then we can return lp fee there.

Reply:
yes, fees are not supported in this simulation release.. we need to come back to it at some point